flocker/provision/_aws.py

```diff
@@ -60,14 +60,16 @@ def _wait_until_running(instance):
     with start_action(
         action_type=u"flocker:provision:aws:wait_until_running",
         instance_id=instance.id,
-    ):
-        while instance.state != 'running':
+    ) as context:
+        while instance.state != u'running' and instance.state != u'terminated':
```

Comment:
Once the state is `shutting-down` is there much hope of this succeeding?  It's a little more complex to look for three states instead of two might it might be worthwhile to make progress faster.

Also ... it should perhaps be even more complicated: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html

I guess "stopping" and "stopped" are technically recoverable... The instance might get started again (and so go back to pending and then running).  But this isn't *supposed* to happen as part of this process.  Likewise for "rebooting".

So it might make sense to only allow "pending" and "running".  Anything else means something unexpected has happened to the instance and making another one is likely the safest thing to do (apart from giving up entirely).

Also, EC2 instances come with `state-reason-code` and `state-reason-message` which tell you what Amazon thinks the reason for the instance being in the state it is in.  This might be interesting information to log as well - perhaps it will give a clue as to why things are in the wrong state.  Off the top of my head, I'm not sure how you get this using boto.  I know that at the AWS EC2 API level, it's part of the `DescribeInstances` response, though.